### ISSUE TITLE
missed softmax

### ISSUE THREAD

**Issue by LDOUBLEV** [AlexHex7/Non-local_pytorch#19]
Great Work!  
However, softmax have missed in this code.  It is also import part for self attention in Non Local neural network.

**Comment by AlexHex7** [comment]
Do you mean the dot product version? I have not tested it yet. 

The paper directly applies 1/N to normalize f to simplify gradient computation and conducts experiments to demonstrate that the performances of different version is similar.

But I think normalization by softmax maybe better than by 1/N. If you are interested in it, you can do this experiment and share the result to me, thanks!

**Comment by JiayongO-O** [comment]
@LDOUBLEV  so have tested the Softmax instead of 1/N for dot product and concatenation version?
